lib/Address.php

```diff
@@ -117,4 +124,13 @@ public function equals($object): bool {
 		return $this->getEmail() === $object->getEmail()
 			&& $this->getLabel() === $object->getLabel();
 	}
+
+	private static function normalizeAddress(string $address): string {
+		// remove single quotes and whitespace the might exist
+		// Examples:
+		// user1@example.com
+		// 'user1@example.com'
+		// ' user1@example.com'
+		return strtolower(trim(trim($address, "'")));
```

Comment:
As mentioned at https://github.com/nextcloud/mail/pull/10980#issuecomment-2787257686.

- `'test@example.org` is a valid email address.
- The current normalizer incorrectly changes it to `test@example.org`

Example: 

```php
if (str_starts_with($address, "'") && str_ends_with($address, "'")) {
	$address = substr($address, 1, -1); // or trim($address, "'"); whatever you prefer
}
return strtolower(trim($address));
```

Given that we need the same logic in the repair job, I'd suggest moving the normalization to a small helper or utility service to make it reusable and more testable.